qiskit_experiments/curve_analysis/data_processing.py

```diff
@@ -61,41 +65,58 @@ def mean_xy_data(
     * ``"iwv"`` *Inverse-weighted variance*
       :math:`\overline{y} = (\sum_{i=1}^N y_i / \sigma_i^2 ) \sigma^2`
       :math:`\sigma^2 = 1 / (\sum_{i=1}^N 1 / \sigma_i^2)`
+    * ``"shots_weighted_variance"`` *Sample mean and variance with weights from shots*
+      :math:`\overline{y} = \sum_{i=1}^N n_i y_i / M`,
+      :math:`\sigma^2 = \sum_{i=1}^N (n_i \sigma_i / M)^2`,
+      where :math:`n_i` is the number of shots per data point and :math:`M = \sum_{i=1}^N n_i`
+      is a total number of shots from different circuit execution at the same x value.
+      If ``shots`` is not provided, this applies uniform weights to all values.
 
     Args
         xdata: 1D or 2D array of xdata from curve_fit_data or
                multi_curve_fit_data
         ydata: array of ydata returned from curve_fit_data or
                multi_curve_fit_data
         sigma: Optional, array of standard deviations in ydata.
+        shots: Optional, array of shots used to get a data point.
         method: The method to use for computing y means and
                 standard deviations sigma (default: "sample").
 
     Returns:
-        tuple: ``(x, y_mean, sigma)`` if ``return_raw==False``, where
+        tuple: ``(x, y_mean, shots, sigma, shots)``, where
```

Comment:
I guess one of the `shots` is to be removed

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/409/commits/d484d65c9f92b450cc03427f694036be3c96df5a